File: carelog-be/src/main/java/carelog/carelog/common/domain/TenantBaseEntity.java
package carelog.carelog.common.domain;

import jakarta.persistence.Column;
import jakarta.persistence.MappedSuperclass;
import lombok.Getter;
import org.hibernate.annotations.Filter;
import org.hibernate.annotations.FilterDef;
import org.hibernate.annotations.ParamDef;

import java.util.UUID;

@Getter
@MappedSuperclass
@FilterDef(
        name = "organizationFilter",
        parameters = @ParamDef(name = "organizationId", type = UUID.class)
)
@Filter(name = "organizationFilter", condition = "organization_id = :organizationId")
public abstract class TenantBaseEntity extends BaseEntity {

    @Column(name = "organization_id", nullable = false, updatable = false)
    private UUID organizationId;

    public void assignOrganization(UUID organizationId) {
        this.organizationId = organizationId;
    }
}

File: src/main/kotlin/carelog/carelog/common/domain/TenantBaseEntity.kt
package carelog.carelog.common.domain

import jakarta.persistence.Column
import jakarta.persistence.MappedSuperclass
import org.hibernate.annotations.Filter
import org.hibernate.annotations.FilterDef
import org.hibernate.annotations.ParamDef
import java.util.UUID

@MappedSuperclass
@FilterDef(
    name = "organizationFilter",
    parameters = [ParamDef(name = "organizationId", type = UUID::class)]
)
@Filter(name = "organizationFilter", condition = "organization_id = :organizationId")
abstract class TenantBaseEntity : BaseEntity() {

    @Column(name = "organization_id", nullable = false, updatable = false)
    var organizationId: UUID? = null
        protected set

    fun assignOrganization(organizationId: UUID) {
        this.organizationId = organizationId
    }
}